composeApp/src/commonMain/kotlin/presentation/feature/home/HomeViewModel.kt

```diff
@@ -4,36 +4,103 @@ import com.rickclephas.kmm.viewmodel.KMMViewModel
 import com.rickclephas.kmm.viewmodel.MutableStateFlow
 import com.rickclephas.kmm.viewmodel.coroutineScope
 import com.rickclephas.kmp.nativecoroutines.NativeCoroutinesState
-import domain.usecase.GetProductsUseCase
-import org.koin.core.component.KoinComponent
-import org.koin.core.component.inject
+import domain.model.Product
+import domain.usecase.favorite.AddToFavoriteUseCase
+import domain.usecase.favorite.GetFavoritesUseCase
+import domain.usecase.favorite.RemoveFromFavoriteUseCase
+import domain.usecase.product.GetProductsUseCase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
 import kotlinx.coroutines.flow.asStateFlow
-import kotlinx.coroutines.flow.onCompletion
-import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
 
 class HomeViewModel : KMMViewModel(), KoinComponent {
     private val getProductsUseCase: GetProductsUseCase by inject()
+    private val getFavoritesUseCase: GetFavoritesUseCase by inject()
+    private val addToFavoriteUseCase: AddToFavoriteUseCase by inject()
+    private val removeFromFavoriteUseCase: RemoveFromFavoriteUseCase by inject()
     private val _state = MutableStateFlow(viewModelScope, HomeScreenState())
 
     @NativeCoroutinesState
     val state = _state.asStateFlow()
 
-    fun onEvent(event: HomeScreenEvent) {
-        when (event) {
-            is HomeScreenEvent.OnLoadMore -> getProducts()
+    fun handleIntent(intent: HomeScreenIntent) {
+        when (intent) {
+            is HomeScreenIntent.OnLaunch -> {
+                viewModelScope.coroutineScope.launch {
+                    getFavorites()
+                    getProducts()
+                }
+            }
+
+            is HomeScreenIntent.OnFavoriteClick -> {
+                val favoriteList = state.value.favoriteList
+
+                if (favoriteList.contains(intent.product))
+                    removeFromFavorite(intent.product)
+                else
+                    addToFavorite(intent.product)
+            }
         }
     }
 
-    private fun getProducts() {
+    private suspend fun getFavorites() {
         viewModelScope.coroutineScope.launch {
-            getProductsUseCase()
-                .onStart { _state.update { it.copy(isLoading = true) } }
-                .onCompletion { _state.update { it.copy(isLoading = false) } }
-                .collect { response ->
-                    _state.update { it.copy(productList = response.products) }
-                }
+            withContext(Dispatchers.IO) {
```

Comment:
Final thing, please remove `withContext`calls from this place to the component that needs it, like a Database. The suspend calls. This will allow us to not care about context switching every time we call a usecase 

Would also be nice if we can inject the dispatcher rather than hardcoding it :)